libdeno/internal.h

```diff
@@ -64,24 +76,6 @@ class DenoIsolate {
   v8::Persistent<v8::ArrayBuffer> shared_ab_;
 };
 
-class UserDataScope {
-  DenoIsolate* deno_;
-  void* prev_data_;
-  void* data_;  // Not necessary; only for sanity checking.
-
- public:
-  UserDataScope(DenoIsolate* deno, void* data) : deno_(deno), data_(data) {
-    CHECK(deno->user_data_ == nullptr || deno->user_data_ == data_);
-    prev_data_ = deno->user_data_;
-    deno->user_data_ = data;
-  }
-
-  ~UserDataScope() {
-    CHECK(deno_->user_data_ == data_);
-    deno_->user_data_ = prev_data_;
-  }
-};
```

Comment:
Generally I'm ok with this - we ought to address the root cause of the issue.

However @piscisaureus debugged this issue - so he I defer approval to him. Bert, when reviewing this have a look at https://github.com/denoland/deno/commit/60c008d23b2bdad333711b43148a5053e83a62cc which may have addressed the original problem.